DBAL/TracingConnectionFactory.php

```diff
@@ -43,15 +43,16 @@ public function createConnection(
         array $mappingTypes = []
     ): Connection {
         $connection = $this->connectionFactory->createConnection($params, $config, $eventManager, $mappingTypes);
+        $reflectionObject = new ReflectionObject($connection);
+        $property = $reflectionObject->getProperty('_conn');
+        $property->setAccessible(true);
+        $previousConnection = $property->getValue($connection);
```

Comment:
Here `$previousConnection` will be null as `connect()` has not been called yet.
This is pretty unsafe to rely on internal connection state anyway.